### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by Setheck**
migrating rpm builder into atsu.
Created centos6 version and added some simple initial tests.

### CODE REVIEW

File: centos/Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+IMAGE_NAME := atsuio/rpmbuilder
+LATEST := $(IMAGE_NAME):latest
+
+.PHONY: build push
+
+build: 
+	docker build -t $(IMAGE_NAME):centos6 -f 6/Dockerfile .
+	docker build -t $(IMAGE_NAME):centos7 -t $(LATEST) -f 7/Dockerfile .
```
Comment (tqorange) [inline centos/Makefile:8]: At some point, consider adding a version number (to support repeatable builds). I also added the version number to as a env var to the container. Not sure if that is a good idea -- could also use a label.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+- sudo apt-get update
+- sudo apt-get install bats
+script:
+- make build
+- make test
+before_deploy:
+- make push
+deploy:
+  provider: releases
+  name: "$GIT_TAG"
```
Comment (tqorange) [inline .travis.yml]: Is this a manual setting?

**Review by tqorange — COMMENTED** [review]
Probably want to figure out what to do the the copyrights:
```
# Copyright 2018 Setheck
#
# This file is part of docker-rpmbuild.
```

```
# Copyright 2015-2016 jitakirin
# Modified by Setheck 6/28/2018
```

etc.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+test:
+	bats **/**/*.bats
```
Comment (tqorange) [inline Makefile:2]: What's the **/** syntax?

File: centos/6/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+# I have forked https://github.com/jitakirin/docker-rpmbuild and modified
+# for my own purposes.
+
+FROM centos:6
+MAINTAINER atsu.io Operations <ops+rpm@atsu.io>
+
+RUN yum install -y rpmdevtools rpm-sign expect yum-utils gpg createrepo_c && \
+    yum clean all && \
+    rm -r -f /var/cache/*
+ADD docker-init.sh docker-rpm-build.sh docker-rpm-sign.sh /usr/local/bin/
```
Comment (tqorange) [inline centos/6/Dockerfile]: COPY? not that it matters here.

File: centos/docker-init.sh [hunk 5]
```diff
@@ -17,59 +17,46 @@
...
+}
+
+DEBUG=false
+BUILD=true
+# parse params
+while [[ "$#" > 0 ]]; do case $1 in
+  --sh)         BUILD=false;shift;;
+  -s|--spec)    SPEC="$2";shift;shift;;
+  -o|--outdir)  OUTDIR="$2";shift;shift;;
+  -a|--sign)    SIGNATURE=(${2//;/ });shift;shift;;
```
Comment (tqorange) [inline centos/docker-init.sh:53]: `-a` seems unintuitive -- is there a reason? Perhaps `-k` for key even though it's more than a key.

File: centos/6/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+# Based on: http://fedoraproject.org/wiki/How_to_create_an_RPM_package
+# And also:
+# - https://registry.hub.docker.com/u/nishigori/rpmbuild
+# - https://registry.hub.docker.com/u/sydneyuni/rpm-build-env/
+#
+# I have forked https://github.com/jitakirin/docker-rpmbuild and modified
+# for my own purposes.
+
+FROM centos:6
```
Comment (tqorange) [inline centos/6/Dockerfile:26]: I do wonder if it's better to deal with the code duplication and a one character change, or template mechanism.

File: centos/6/rpmbuilder-c6.bats [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bats
```
Comment (tqorange) [inline centos/6/rpmbuilder-c6.bats:1]: Is `bats` a widely used tool?

**Review by tqorange — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (Setheck) [reply Makefile:2]: this is just file globbing `**` is to look into any directory

File: centos/6/Dockerfile (hunk 4 above)
Comment (Setheck) [reply centos/6/Dockerfile]: sure, i'll update

File: centos/docker-init.sh (hunk 5 above)
Comment (Setheck) [reply centos/docker-init.sh:53]: `s` was taken. I was thinking a like auth. But I'm fine with `k` if you feel it more appropriate.

File: centos/6/rpmbuilder-c6.bats (hunk 7 above)
Comment (Setheck) [reply centos/6/rpmbuilder-c6.bats:1]: I don't think bash testing is very popular, I haven't seen any alternatives to bats but i'm open to suggestions.

File: .travis.yml (hunk 2 above)
Comment (Setheck) [reply .travis.yml]: Could be, I wasn't sure how we would version releases. I guess I could still generate a sqrl tag for the github release?

File: centos/Makefile (hunk 1 above)
Comment (Setheck) [reply centos/Makefile:8]: added to card about this

File: centos/6/Dockerfile (hunk 6 above)
Comment (Setheck) [reply centos/6/Dockerfile:26]: yeah, i'm open to change if it makes more sense in the future

**Pull request merged**